packages/component-test-utils-react/src/shallow.spec.js

```diff
@@ -146,5 +146,64 @@ describe('react shallow render', () => {
 
       expect(cmp.html()).toBe('<button type="button">0</button>');
     });
+
+    it('should render event with anonymous function handler', () => {
+      const Component = () => {
+        const [nbLikes, setNbLike] = React.useState(0);
+        return (
+          <button type="button" onClick={() => setNbLike(nbLikes + 1)}>
+            {nbLikes}
+          </button>
+        );
+      };
+
+      const cmp = shallow(<Component/>);
+
+      expect(cmp.html()).toBe(
+        '<button type="button" onClick="[onClick]">0</button>'
+      );
+    });
+
+    it('should render event with named handler', () => {
+      const Component = () => {
+        const [nbLikes, setNbLike] = React.useState(0);
+        function clickHandler() {
+          setNbLike(nbLikes + 1);
+        }
+
+        return (
+          <button type="button" onClick={clickHandler}>
+            {nbLikes}
+          </button>
+        );
+      };
+
+      const cmp = shallow(<Component/>);
+
+      expect(cmp.html()).toBe(
+        '<button type="button" onClick="[clickHandler]">0</button>'
+      );
+    });
+  });
+
+  describe('dispatchEvent', () => {
+    it('should trigger click event on the button and update state', () => {
+      const Component = () => {
+        const [nbLikes, setNbLike] = React.useState(0);
+        return (
+          <button type="button" onClick={() => setNbLike(nbLikes + 1)}>
+            {nbLikes}
+          </button>
+        );
+      };
+
+      const cmp = shallow(<Component/>);
+
+      cmp.dispatchEvent('Click');
```

Comment:
For the DX, WYT of `cmp.make('click')` ? Or something that is more readable that `dispatchEvent` ?